### ISSUE TITLE
E4.1-S3: Add released-artifact ONNX first-crack detector backend

### ISSUE THREAD

**Issue by syamaner** [syamaner/coffee-roaster-mcp#106]
## Goal
Add the runtime detector backend that consumes the released Hugging Face ONNX artifacts already resolved by Epic 4, without adding training, export, or sync behavior.

## Done when
- `first_crack.mode: audio` can construct a detector backend from the resolved ONNX model and feature-extractor config.
- INT8 and FP32 precision selections use the existing E4 resolver/validation boundary.
- Missing or invalid runtime dependencies/artifacts fail clearly before detection starts.
- Tests use small fakes or mocks for the ONNX runtime path and do not require model downloads or real ONNX files in CI.
- Runtime metadata includes precision, repo id, revision, model filename, feature extractor filename, and confidence when available.

## Out of scope
- Model training, ONNX export, Hugging Face sync, detector quality evaluation, audio capture lifecycle, and MCP session wiring.

Parent epic: #103

**Comment by syamaner** [comment]
Implemented E4.1-S3 on branch `feature/106-add-released-artifact-onnx-first-crack-detector-backend`.

What changed:
- Added released-artifact ONNX detector backend construction for `first_crack.mode: audio`.
- Resolves configured INT8/FP32 artifacts through the existing Hugging Face/local artifact resolver boundary.
- Loads the precision-specific `preprocessor_config.json` through `transformers.ASTFeatureExtractor` and creates an ONNX Runtime CPU session with configured thread limits.
- Adapts ONNX logits into the existing detector output/event metadata path with first-crack confidence.
- Added clear fail-fast errors for invalid mode, invalid thread config, missing/invalid preprocessor config, missing runtime dependencies, missing model inputs, empty outputs, and sample-rate mismatch.
- Updated durable state docs and README notes.

Validation:
- `./.venv/bin/python -m pytest tests/test_detector.py tests/test_artifacts.py`: 43 passed.
- `./.venv/bin/python -m pytest --cov=coffee_roaster_mcp --cov-report=term-missing:skip-covered --cov-report=json:coverage.json --cov-report=html:htmlcov`: 272 passed, coverage 90.45%.
- `./.venv/bin/python -m ruff check .`: passed.
- `./.venv/bin/python -m ruff format --check .`: passed.
- `./.venv/bin/python -m pyright`: 0 errors.

Kept out of scope: session-owned detector startup, audio lifecycle wiring, automatic T0, rolling metrics, final log schemas, training/export/HF sync, real microphone validation, live Hottop validation, and broad release validation.
